polling2.py

```diff
@@ -180,15 +180,16 @@ def poll(
     This message should allow a user to work-out how long the poll could take, and thereby detect a hang in real-time
     if the poll takes longer than it should.
     """
-
-    assert (timeout is not None or max_tries is not None) or poll_forever, (
-        "You did not specify a maximum number of tries or a timeout. Without either of these set, the polling "
-        'function will poll forever. If this is the behavior you want, pass "poll_forever=True"'
-    )
-
-    assert not (
-        (timeout is not None or max_tries is not None) and poll_forever
-    ), "You cannot specify both the option to poll_forever and max_tries/timeout."
+    if not ((timeout is not None or max_tries is not None) or poll_forever):
+        raise AssertionError(
```

Comment:
Since we're including this in a major release, I don't think it would be unreasonable to break compatibility by switching to a `ValueError`.

(FWIW, I don't actually need/rely on this at all. It's just more correct IMO.)

Alternatively, a compromise solution could be to create a custom exception for a while that inherits from both:
``` python
class ValueAssertionError(ValueError, AssertionError):
    """Backwards-compatible ValueError (catching AssertionError is deprecated!)"""
```